DataProcessing/CryptoCoarseFundamentalUniverseDataDownloader.cs

```diff
@@ -0,0 +1,283 @@
+            _fromDate = fromDate;
+            _toDate = toDate;
+
+            Directory.CreateDirectory(_destinationFolder);
+        }
+
+        /// <summary>
+        /// Load the local base trade data files, convert into universe file
+        /// </summary>
+        public bool ConvertToUniverseFile()
+        {
+            try
+            {
+                // Get base currency for reference of USD dollar volume conversion
+                var baseCurrency = new Dictionary<string, string>();
+
+                var symbolProperties = ReadOnlineDocumentLines("https://raw.githubusercontent.com/QuantConnect/Lean/master/Data/symbol-properties/symbol-properties-database.csv");
+                foreach (var line in symbolProperties)
+                {
+                    var lineItem = line.Split(",");
+                    if (lineItem.First() == _market)
+                    {
+                        baseCurrency[lineItem[1]] = lineItem[4];
+                    }
+                }
+
+                var files = new List<string>();
+
+                // Get all trade data files in set crypto market data file
+                foreach(var file in Directory.GetFiles(Path.Combine(_baseFolder, "daily"), "*_trade.zip", SearchOption.AllDirectories))
+                {
+                    files.Add(file);
+                }
+
+                foreach (var fileString in files)
+                {
+                    // Fetch price data from zip files
+                    using (ZipArchive zip = ZipFile.Open(fileString, ZipArchiveMode.Read))
+                    {
+                        foreach (ZipArchiveEntry entry in zip.Entries)
+                        {
+                            string name = entry.Name.Split(".").First().ToUpper();
+
+                            var streamFile = entry.Open();
+                            var lines = ReadLines(streamFile);
+
+                            foreach (var line in lines)
+                            {
+                                var csv = line.Split(",");
+                                var dateTime = Parse.DateTimeExact(csv[0].Split(" ").First(), "yyyyMMdd");
+                                if (dateTime < _fromDate || dateTime > _toDate)
+                                {
+                                    continue;
+                                }
+
+                                var date = $"{dateTime:yyyyMMdd}";
+                                var high = decimal.Parse(csv[2], NumberStyles.Any, CultureInfo.InvariantCulture);
+                                var low = decimal.Parse(csv[3], NumberStyles.Any, CultureInfo.InvariantCulture);
+                                var price = decimal.Parse(csv[4], NumberStyles.Any, CultureInfo.InvariantCulture);
+                                var volume = decimal.Parse(csv[5], NumberStyles.Any, CultureInfo.InvariantCulture);
+                                var dollarVolume = price * volume;
```

Comment:
Could replace by using existing `LeanDataReader` I think, avoid all the zip and parsing entirely